source/include/dqm4hep/Internal.h

```diff
@@ -100,6 +100,11 @@ namespace dqm4hep {
     class QualityTest;
     class QualityTestReport;
     class Path;
+    class Plugin;
+    template <typename T>
+    class Directory;
+    template <typename T>
```

Comment:
Why do you need the `template <typename T>` here?

Reply:
For the typedefs a few lines after

Reply:
Seems to me that the type def don't need the `typename T` actually.